### ISSUE TITLE
GLES 2 screen jank 

### ISSUE THREAD

**Issue by gianmichelemariani** [godotengine/godot#33873]
**Godot version:**
Godot 3.2 beta 2

**OS/device including version:**
OSX 10.14.6
Macbook Pro 13" 2017. Intel Iris 640

**Issue description:**
When using GLES2 on a simple project, I get a serious jank/stutter even on a simple project when running from the editor.

https://www.dropbox.com/s/c5fcxoh65ub65c2/godot_jank_es2.mov?dl=0

If I switch this project to GLES3 it runs butter smooth, as I would expect.

Now, if I export the project as an osx app and run, it is also butter smooth, which makes me think it is something related to running from the editor.
In fact if I still keep the editor as GLES3, but I force GLES2 rendering via project settings, there is no jank/jitter.

<img width="1040" alt="Screenshot 2019-11-25 at 08 08 19" src="https://user-images.githubusercontent.com/46694168/69523271-5c1bfc80-0f5b-11ea-9daa-576d52e82356.png">

Unfortunately when I reopen the project this switches the editor automatically to GLES2 and the jitter persists.

Anyone else had this problem?

I've tried to disable all debug/stout options with no luck

**Steps to reproduce:**
Just build the project in GLES2 on OSX

**Minimal reproduction project:**
[simple_test.zip](https://github.com/godotengine/godot/files/3885425/simple_test.zip)

**Comment by gianmichelemariani** [comment]
Kept experiment with this  as I have switched to OSX 10.15.2

The problem still persists, but I have also found that when play the project (command+b), if I minimize the editor or set the game in full screen, the jank stops. As soon as the editor is back on the screen, stuttering starts again.

**Comment by Calinou** [comment]
@gianmichelemariani This could be solved by making the editor not re-render anything while the project is running if it's not focused. However, since this can make debugging a bit harder, it should probably be a setting that's disabled by default.

**Comment by gianmichelemariani** [comment]
I wonder what's different with gles3 projects that run perfectly well.

**Comment by hmans** [comment]
I have the same issues with my current (GLES2) project -- I can hardly run the thing in Godot for macOS. (I'm generally seeing very poor editor performance on macOS with the integrated Intel Graphics, documented in #17584, which this issue here may or may not be related to.)

One thing I noticed was that there's _specifically_ noticeable jank when the game creates instances of preloaded scenes. I've tried if the profiler surfaced something, but it didn't.
